### PR TITLE
Chem Comp to Staging

### PR DESCRIPTION
**Pull request by santiblaumann**
Since the previous PR, bug fixes to both the fluent syntax and the test cases themselves have been made. Provided this PR has no issues, chemical attribute search should be fully implemented.

### CODE REVIEW

File: rcsbsearchapi/search.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ class Terminal(Query):
     attribute: Optional[str] = None
     operator: Optional[str] = None
     value: Optional[TValue] = None
-    service: str = "text"
+    service: str = "text"  # eventually this should be based on whether or not search is chem or struct
```
[the remark this answers is not in the document]
Comment (piehld) [reply rcsbsearchapi/search.py:211]: Or, if the default needs to be assigned up here, then I would at least change `"text"` to the be the constant variable, as you use on line 407 below.

File: rcsbsearchapi/search.py [hunk 2]
```diff
@@ -423,43 +424,43 @@ def contains_words(
             value = value.value
         if isinstance(value, list):
             value = " ".join(value)
-        return Terminal(self.attribute, "contains_words", value)
+        return Terminal(self.attribute, "contains_words", value, self.type)
```
Comment (piehld) [inline rcsbsearchapi/search.py:427]: And here...and elsewhere of course. Sorry again.

File: rcsbsearchapi/update_schema.py
Comment (piehld) [inline rcsbsearchapi/update_schema.py:1]: Can you actually keep this file here? I think we may find it useful later to make downloading and updating the schema resource files easier (we'll have to update this though, of course).

**Review by piehld — CHANGES_REQUESTED** [review]
Thanks @santiblaumann! Excellent work getting the chemical attribute search enabled. I just had a few suggestions on some changes to make before merging, which I comment on below.

FYI @brindakv

File: rcsbsearchapi/update_schema.py
Comment (piehld) [reply rcsbsearchapi/update_schema.py:1]: (Just to clarify, don't worry about updating the file now—we can do that later. For now, just add the file back in.)

File: rcsbsearchapi/const.py [hunk 3]
```diff
@@ -1 +1,8 @@
 REQUESTS_PER_SECOND = 10
+STRUCTURE_INDEX = 0
+CHEMICAL_INDEX = 0
+METADATA_SCHEMA_URL = "http://search.rcsb.org/rcsbsearch/v2/metadata/schema"
+CHEMICAL_SCHEMA_URL = "https://search.rcsb.org/rcsbsearch/v2/metadata/chemical/schema"
+SEARCH_SCHEMA_URL = "http://search.rcsb.org/json-schema-rcsb_search_query.json"
+STRUCTURE = "text"
+CHEMICAL = "text_chem"
```
[the remark this answers is not in the document]
Comment (santiblaumann) [reply rcsbsearchapi/const.py:8]: I think the first suggestion is a little too verbose. Could I instead propose "STRUCTURE_ATTRIBUTE_SEARCH"  and "CHEMICAL_ATTRIBUTE_SEARCH"?

File: rcsbsearchapi/const.py [hunk 4]
```diff
@@ -1 +1,8 @@
 REQUESTS_PER_SECOND = 10
+STRUCTURE_INDEX = 0
+CHEMICAL_INDEX = 0
+METADATA_SCHEMA_URL = "http://search.rcsb.org/rcsbsearch/v2/metadata/schema"
+CHEMICAL_SCHEMA_URL = "https://search.rcsb.org/rcsbsearch/v2/metadata/chemical/schema"
+SEARCH_SCHEMA_URL = "http://search.rcsb.org/json-schema-rcsb_search_query.json"
```
[the remark this answers is not in the document]
Comment (piehld) [reply rcsbsearchapi/const.py:6]: Sorry, `SEARCH_SCHEMA_URL` should be: `https://search.rcsb.org/schema/search/request/json-schema-rcsb_search_query.json`. But can you leave the other URL in there too (https://search.rcsb.org/openapi.json)? Just assign it to a new constant variable (e.g., `SEARCH_OPENAPI_SCHEMA_URL`).

**Comment by piehld** [comment]
Another constant I would add: `FULL_TEXT_SEARCH_SERVICE = "full_text"`. Then replace it where used (I think it's only used once).

File: rcsbsearchapi/search.py [hunk 5]
```diff
@@ -154,15 +154,15 @@ def and_(self, other: Union[str, "Attr"]) -> "PartialQuery":
         ...
 
     def and_(
-        self, other: Union[str, "Query", "Attr"]
+        self, other: Union[str, "Query", "Attr"], qtype="text"
```
Comment (piehld) [inline rcsbsearchapi/search.py:157]:
```suggestion
        self, other: Union[str, "Query", "Attr"], qtype=STRUCTURE_ATTRIBUTE_SEARCH_SERVICE
```

File: rcsbsearchapi/search.py [hunk 6]
```diff
@@ -174,14 +174,14 @@ def or_(self, other: "Query") -> "Query":
     def or_(self, other: Union[str, "Attr"]) -> "PartialQuery":
         ...
 
-    def or_(self, other: Union[str, "Query", "Attr"]) -> Union["Query", "PartialQuery"]:
+    def or_(self, other: Union[str, "Query", "Attr"], qtype="text") -> Union["Query", "PartialQuery"]:
```
Comment (piehld) [inline rcsbsearchapi/search.py:177]:
```suggestion
    def or_(self, other: Union[str, "Query", "Attr"], qtype=STRUCTURE_ATTRIBUTE_SEARCH_SERVICE) -> Union["Query", "PartialQuery"]:
```

File: tests/testschema.py [hunk 7]
```diff
@@ -52,11 +53,23 @@ def testSchema(self):
...
+        ok = webSchema == localSchema
         self.assertTrue(ok)
-        logger.info("Schema tests results: local version (%r) and web version (%s)", local_schema.get("$comment"), web_schema.get("$comment"))
+        localVer = localSchema.get("$comment")
+        webVer = webSchema.get("$comment")
+        logger.info("Metadata schema tests results: local version (%r) and web version (%s)", localVer, webVer)
+        webSchema = _download_schema(CHEMICAL_ATTRIBUTE_SCHEMA_URL)
+        localSchema = _load_chem_schema()
+        # logger.info("webschema: %s", webSchema)
+        # logger.info("localSchema: %s", localSchema)
```
Comment (piehld) [inline tests/testschema.py:66]:
```suggestion
```

File: notebooks/covid.ipynb
Comment (piehld) [inline notebooks/covid.ipynb:1]: Can you remove the `print(attrs)` from the first cell?

**Review by piehld — COMMENTED** [review]
Thanks for all the quick changes @santiblaumann! Just a few more tiny suggestions (you should be able to just click "accept suggestion" in the GitHub UI to create the commit for them.

**Review by piehld — APPROVED** [review]

**Pull request merged**